lib/langchain.rb

```diff
@@ -28,43 +42,35 @@ module Processors
     autoload :PDF, "langchain/processors/pdf"
     autoload :Text, "langchain/processors/text"
   end
-end
 
-module Agent
-  autoload :Base, "agent/base"
-  autoload :ChainOfThoughtAgent, "agent/chain_of_thought_agent/chain_of_thought_agent.rb"
-end
+  module Prompt
+    require_relative "langchain/prompt/loading"
 
-module Vectorsearch
-  autoload :Base, "vectorsearch/base"
-  autoload :Chroma, "vectorsearch/chroma"
-  autoload :Milvus, "vectorsearch/milvus"
-  autoload :Pinecone, "vectorsearch/pinecone"
-  autoload :Pgvector, "vectorsearch/pgvector"
-  autoload :Qdrant, "vectorsearch/qdrant"
-  autoload :Weaviate, "vectorsearch/weaviate"
-end
-
-module LLM
-  autoload :Base, "llm/base"
-  autoload :Cohere, "llm/cohere"
-  autoload :GooglePalm, "llm/google_palm"
-  autoload :HuggingFace, "llm/hugging_face"
-  autoload :OpenAI, "llm/openai"
-  autoload :Replicate, "llm/replicate"
-end
+    autoload :Base, "langchain/prompt/base"
+    autoload :PromptTemplate, "langchain/prompt/prompt_template"
+    autoload :FewShotPromptTemplate, "langchain/prompt/few_shot_prompt_template"
+  end
 
-module Prompt
-  require_relative "prompt/loading"
+  module Tool
+    autoload :Base, "langchain/tool/base"
+    autoload :Calculator, "langchain/tool/calculator"
+    autoload :SerpApi, "langchain/tool/serp_api"
+    autoload :Wikipedia, "langchain/tool/wikipedia"
+  end
 
-  autoload :Base, "prompt/base"
-  autoload :PromptTemplate, "prompt/prompt_template"
-  autoload :FewShotPromptTemplate, "prompt/few_shot_prompt_template"
+  module Vectorsearch
+    autoload :Base, "langchain/vectorsearch/base"
+    autoload :Chroma, "langchain/vectorsearch/chroma"
+    autoload :Milvus, "langchain/vectorsearch/milvus"
+    autoload :Pinecone, "langchain/vectorsearch/pinecone"
+    autoload :Pgvector, "langchain/vectorsearch/pgvector"
+    autoload :Qdrant, "langchain/vectorsearch/qdrant"
+    autoload :Weaviate, "langchain/vectorsearch/weaviate"
+  end
 end
 
-module Tool
-  autoload :Base, "tool/base"
-  autoload :Calculator, "tool/calculator"
-  autoload :SerpApi, "tool/serp_api"
-  autoload :Wikipedia, "tool/wikipedia"
-end
+Agent = Langchain::Agent
+LLM = Langchain::LLM
+Prompt = Langchain::Prompt
+Tool = Langchain::Tool
+Vectorsearch = Langchain::Vectorsearch
```

Comment:
I don't think we need these constants. I would just have to bump the version from 0.3.x to 0.4.x to indicate the breaking changes.